### PR TITLE
Using the latest device-agnostic api in Context

### PR DESCRIPTION
**Pull request by FFFrog**
There are some code refactorings for Context in PyTorch repo, refer to [this](https://github.com/pytorch/pytorch/pull/136519) to get more info, we want to merge this pr in torch-xpu-ops first and then update the pin-commit in PyTorch to make the ci workflow green in PyTorch.

### CODE REVIEW

File: src/ATen/native/xpu/sycl/ResizeKernel.cpp [hunk 1]
```diff
@@ -25,7 +25,7 @@ void resize_bytes_xpu(StorageImpl* storage, size_t size_bytes) {
   c10::xpu::XPUGuard guard(device.index());
   at::DataPtr data = allocator->allocate(size_bytes);
   if (storage->data_ptr()) {
-    at::globalContext().lazyInitXPU();
+    at::globalContext().lazyInitDevice(c10::DeviceType::XPU);
```
Comment (EikanWang) [inline src/ATen/native/xpu/sycl/ResizeKernel.cpp:28]: I suppose this line is invalid. @fengyuan14 , @guangyey , why do we require `lazyInitDevice`?

Comment (guangyey) [reply src/ATen/native/xpu/sycl/ResizeKernel.cpp:28]: I agree with your point of view. `lazyInitXPU` should already be called in [`_lazy_init`](https://github.com/pytorch/pytorch/blob/e14b58ffbd4758fb37a4118efb3f01e6a47e0ff2/torch/xpu/__init__.py#L120).

Comment (guangyey) [inline src/ATen/native/xpu/sycl/ResizeKernel.cpp:28]:
```suggestion
```

Comment (FFFrog) [reply src/ATen/native/xpu/sycl/ResizeKernel.cpp]: Thank you, I will remove this unnecessary init right now

Comment (FFFrog) [reply src/ATen/native/xpu/sycl/ResizeKernel.cpp]: Done

**Comment by guangyey** [comment]
Thanks very much~

**Review by guangyey — APPROVED** [review]

**Pull request merged**